libbeat/scripts/Makefile

```diff
@@ -417,8 +417,7 @@ env-logs:
 	${DOCKER_COMPOSE} logs -f
 
 fix-permissions:
-	# Change ownership of all files inside /build folder from root/root to current user/group
-	docker run -v ${PWD}:/beat alpine:3.4 sh -c "find /beat -user 0 -exec chown -h $(shell id -u):$(shell id -g) {} \;"
+	sh ${ES_BEATS}/script/fix_permissions.sh "${PWD}"
```

Comment:
Maybe it would make sense to move this to `mage`, in order to create a more platform independent build system? Currently the x-pack Makefile does not implement `fix-permissions`. Do we need this change for them as well?

Reply:
This particular fix was a way to help the CI pipeline to be configured as expected when reusing CI Workers in order to speed up the overall execution time.

Unfortunately, I've no clue if that's something to be used by the x-pack Makefiles. The sh script is the one called from the CI Pipeline.

Regarding to move this to `mage`, I've no clue what bits and pieces I do need to change, this is actually a bit less intrusive way I've found so far. Other than that, is that something we can potentially do in a follow up? Or maybe someone else could jump in and help with?

Thanks

Reply:
Looking into this (Thanks @andrewkroh), this PR looks ok. The magefiles do the cleanup automatically, this is why `fix-permissions` is empty for the x-pack Makefile. We should be able to remove `fix-permissions` from the x-pack Makefiles.

Reply:
Thanks for reviewing this @urso .  We we ok with moving forward with this PR & then open a new PR to remove `fix-permissions` from the x-pack Makefiles.

Reply:
Before removing the `fix-permissions` from `dev-tools/make/xpack.mk` check if Jenkins in configured somewhere to directly invoke `make fix-permissions` anywhere. If it is then we'll need to keep it from compatibility for now.

Reply:
checking the job definition on the infra repository, there is no reference to `'fix-permissions` on any job definition file.
`grep -ri 'fix-permissions' ci/jjb/beats-ci`

there is a reference on one of the scripts on dev-tools folder in the beats repo

```
>grep -ri 'fix-permissions' dev-tools/

dev-tools//make/xpack.mk:fix-permissions:
dev-tools//jenkins_ci.sh:    make fix-permissions || true
```

I did not find any other reference

```
grep -ri 'fix-permissions' .ci/
grep -ri 'fix-permissions' script/
```

Reply:
@andrewkroh I see the following instances of `fix-permissions` in the beats repo.

```
libbeat/scripts/Makefile
303:		$(MAKE) fix-permissions; \
419:fix-permissions:

dev-tools/make/xpack.mk
27:fix-permissions:

dev-tools/jenkins_ci.sh
25:    make fix-permissions || true
```

It looks like we need leave it for compatibility, correct?

Reply:
Let's keep it in for now. We want to switch the OSS to mage as well (which is already cleaner than what we have in our makefiles).

Reply:
Thanks @urso & @andrewkroh , are you fine with proceeding with the PR as it stands?

Reply:
We'll move forward with this PR & address any unrelated failures in the beats pipeline.